File: backend/src/main/java/com/example/entity/Post.java
package com.example.entity;

import java.util.Date;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import com.example.entity.vo.TimeSlot;

import lombok.Data;

@Data
@TableName("post")
public class Post {
    @TableId(type = IdType.AUTO)
    private Integer id;
    private Integer userId;
    private String title;
    private String content;
    private String type;  // help求助/resource资源
    private Integer subjectId;
    private Integer placeId;
    
    @TableField(typeHandler = JacksonTypeHandler.class)
    private TimeSlot timeSlot;
    
    private Integer inviteEnabled;
    private Integer status;
    private Date createTime;
    private Date updateTime;
}

File: fontend/src/net/user/post.js
// 帖子相关

import { get, post, put, del } from '../index'

/**
 * 获取帖子列表 
 * @param success 成功回调
 */
export function getPostList(success) {
    get('/api/post/', success)
}

/**
 * 获取帖子详情
 * @param id 帖子ID
 * @param success 成功回调
 */
export function getPostById(id, success) {
    get(`/api/post/${id}`, success)
}

/**
 * 发布帖子
 * @param data {title:标题, content:内容, type:类型(help/resource), subjectId:科目ID, placeId:场地ID, timeSlot:时间段}
 * @param success 成功回调
 */
export function createPost(data, success) {
    post('/api/post/', data, success)
}

/**
 * 更新帖子
 * @param data {id:帖子ID, title:标题, content:内容, type:类型, subjectId:科目ID, placeId:场地ID, timeSlot:时间段}
 * @param success 成功回调
 */
export function updatePost(data, success) {
    put('/api/post/', data, success)
}

/**
 * 删除帖子(实际是关闭)
 * @param id 帖子ID
 * @param success 成功回调
 */
export function deletePost(id, success) {
    del(`/api/post/${id}`, success)
}